ra2ce/analysis/analysis_factory.py

```diff
@@ -51,20 +69,105 @@ def get_analysis(self, analysis_config: AnalysisConfigWrapper) -> AnalysisProtoc
             NotImplementedError: The analysis type is not implemented.
 
         Returns:
-            AnalysisProtocol: The analysis to be executed.
+            AnalysisIndirectProtocol: The direct analysis to be executed.
         """
+        _input_dict = dict(
+            analysis=self.analysis,
+            input_path=analysis_config.config_data.input_path,
+            output_path=analysis_config.config_data.output_path,
+        )
         if self.analysis.analysis == AnalysisDirectEnum.DIRECT:
             return DirectDamage(
-                analysis_config.graph_files.base_network_hazard,
-                self.analysis,
-                analysis_config.config_data.input_path,
-                analysis_config.config_data.output_path,
+                graph_file=analysis_config.graph_files.base_network_hazard,
```

Comment:
why not including this in the above dictionary? you are using the same property, right?

Reply:
In these cases it is indeed the same, but in general the graph is different between the analysis, which I want to make explicit.